evdev/uinput.c

```diff
@@ -97,6 +97,24 @@ uinput_set_prop(PyObject *self, PyObject *args)
         return NULL;
 }
 
+static PyObject *
+uinput_get_sysname(PyObject *self, PyObject *args)
+{
+    int fd;
+    char sysname[64];
+
+    int ret = PyArg_ParseTuple(args, "i", &fd);
+    if (!ret) return NULL;
+
+    if (ioctl(fd, UI_GET_SYSNAME(sizeof(sysname)), &sysname) < 0)
+        goto on_err;
+
+    return Py_BuildValue("s", &sysname);
+
+    on_err:
+        PyErr_SetFromErrno(PyExc_IOError);
```

Comment:
`PyExc_IOError` seems to be the same thing as `PyExc_OSError` (see https://docs.python.org/3/c-api/exceptions.html), however, I was confused at first and had to look it up. Can we use `PyExc_OSError` here instead?

Reply:
I think we can. It seems that OSError and IOError are aliases in modern Python, with the latter existing only for backwards compatibility:

```
$ python -V
Python 3.11.6
$ python -c "print(OSError == IOError)"
True
```

This seems to have been different in the past and was changed with [PEP 3151](https://peps.python.org/pep-3151/).

The entirety of `input.c` and `uinput.c` appear to be using IOError. If `uinput_get_sysname()` were to be the only function using `OSError`, then this function would be the odd one that does not follow the style of the rest. On the other hand, I do not know if there are unforeseen consequences on old versions of Python if I replace all IOErrors of all functions with OSErrors.

(I guess that this function is a bit of an odd one anyway since most other Python functions don't attempt to catch errors thrown by C functions they call.)

Maybe the Python function `_find_device()` itself should catch IOError instead of OSError? I only picked OSError because I ran Python to see what would happen if `uinput_get_uinput` failed, saw that it threw OSError, and thus decided to catch OSError. Maybe the decision to throw IOError but catch OSError can lead to compatibility problems on old Python versions?

Do we have a policy on the minimum Python version we support?

Reply:
> On the other hand, I do not know if there are unforeseen consequences on old versions of Python if I replace all IOErrors of all functions with OSErrors.
> Maybe the decision to throw IOError but catch OSError can lead to compatibility problems on old Python versions?

3.3 has reached end of life a long time ago https://devguide.python.org/versions/, I think this should be fine. Someone using a system that outdated will probably not suddenly update python-evdev.

Reply:
I've added a commit to replace all mentions of IOError with OSError in the C bindings. OSError seems to be the preferred name on new Python versions. This removes the confusion of throwing IOError and catching OSError.